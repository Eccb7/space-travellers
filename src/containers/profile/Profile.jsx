import React from 'react';
import { useSelector } from 'react-redux';
import './profile.css';
import { MissionProfile } from '../../components';

function Profile() {
  const rockets = useSelector((state) => state.rockets.Data.filter((rocket) => rocket.reserved));
  const missions = useSelector((state) => state.mission.missions
    .filter((mission) => mission.reserved));

  return (
    <div className="my-profile-container">
      <h2>My Profile</h2>
      <div className="sectionTitle">My Rockets</div>
      {rockets.length === 0 ? (
        <p>No rockets reserved yet!</p>
      ) : (
        <table className="holderTable">
          <tbody>
            {rockets.map((rocket) => (
              <tr key={rocket.id} className="rocketrow">
                <td className="rocketdata">{rocket.name}</td>
              </tr>
            ))}
          </tbody>
        </table>
      )}
      {missions.length === 0 ? (<p>no missions joined</p>)
        : (
          <table className="holderTable">
            <thead>
              <tr className="app_mission_profile-row">
                <th className="app_mission_profile-column">My Missions</th>
              </tr>
            </thead>
            <tbody>
              {
            missions.map((mission) => (
              <MissionProfile key={mission.mission_id} missionName={mission.mission_name} />
            ))
          }
            </tbody>
          </table>
        )}
    </div>
  );
}

export default Profile;
