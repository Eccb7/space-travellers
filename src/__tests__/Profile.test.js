import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Profile from '../containers/profile/Profile';

const mockStore = configureStore();

test('matches snapshot', () => {
  const store = mockStore({
    rockets: { Data: [] },
    mission: { missions: [] },
  });

  const { asFragment } = render(
    <Provider store={store}>
      <Profile />
    </Provider>,
  );

  expect(asFragment()).toMatchSnapshot();
});
