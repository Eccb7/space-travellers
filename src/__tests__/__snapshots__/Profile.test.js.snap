// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="my-profile-container"
  >
    <div>
      <span
        class="sectionTitle"
      >
        My Rockets
      </span>
      <p>
        No rockets reserved yet!
      </p>
    </div>
    <div>
      <span
        class="sectionTitle"
      >
        My Missions
      </span>
      <p>
        No missions joined
      </p>
    </div>
  </section>
</DocumentFragment>
`;
